src/briefcase/integrations/virtual_environment.py

```diff
@@ -0,0 +1,235 @@
+
+    @property
+    def executable(self) -> str:
+        """Path to the Python executable in the virtual environment.
+
+        :return: Absolute filesystem path to the venv's Python executable.
+        """
+        python = self.bin_dir / ("python.exe" if os.name == "nt" else "python")
+        return os.fspath(python)
+
+    def exists(self) -> bool:
+        """Check whether the virtual environment exists.
+
+        :return True if venv exists and contains a pyvenv.cfg file. else False
+        """
+        return self.venv_path.exists() and (self.venv_path / "pyvenv.cfg").exists()
+
+    def create(self) -> None:
+        """Create the virtual environment.
+
+        :raises BriefcaseCommandError if creation fails.
+        """
+        try:
+            self.venv_path.parent.mkdir(parents=True, exist_ok=True)
+            self.tools.subprocess.run(
+                [sys.executable, "-m", "venv", os.fspath(self.venv_path)],
+                check=True,
+            )
+            self.update_core_tools()
+        except subprocess.CalledProcessError as e:
+            raise BriefcaseCommandError(
+                f"Failed to create virtual environment at {self.venv_path}"
+            ) from e
+
+    def recreate(self) -> None:
+        """Remove and re-create the virtual environment."""
+        if self.exists():
+            shutil.rmtree(self.venv_path)
+        self.create()
+
+    def update_core_tools(self) -> None:
+        """Upgrade core Python packaging tooling in the venv."""
+        try:
+            self.run([self.executable, "-m", "pip", "install", "-U", "pip"], check=True)
+        except Exception as e:
+            raise BriefcaseCommandError(
+                f"Failed to bootstrap pip tooling at {self.venv_path}"
+            ) from e
+
+    def _rewrite_head(self, args: SubprocessArgsT) -> SubprocessArgsT:
+        """Rewrite the first argument to ensure it points to the venv's Python
+        executable.
+
+        param args: Original subprocess arguments.
+        :return: Updated subprocess arguments.
+        """
+        if not args:
+            return args
+        head = os.fspath(args[0])
+        if os.path.normcase(head) == os.path.normcase(sys.executable):
+            return [self.executable, *args[1:]]
+
+        if os.path.normcase(head) == os.path.normcase(sys.executable):
+            return [self.executable, *args[1:]]
+        return list(args)
+
+    def full_env(self, overrides: dict[str, str | None] | None) -> dict[str, str]:
+        """Generate the full environment for the venv.
+
+        :param overrides: Environment variables to override or unset. Can be None if
+            there are no explicit environment changes. Use None values to unset
+            environment variables
+        :return: environment mapping for the venv with overrides applied.
+        """
+
+        env = {}
+
+        if overrides:
+            env.update({k: v for k, v in overrides.items() if v is not None})
```

Comment:
This won't *quite* work as expected.

Briefcase is doing a little trick here; all environment values *should* be strings; but if you pass in a value of `None`, this is interpreted as "make sure this value is *cleared* in the final environment". This is mostly used for flags like `PIP_REQUIRE_VIRTUALENV`, which the end user might have in their environment, but will interfere will how Briefcase is operating.

That final "clear" needs to happen against the *final* environment, so it needs to be preserved for the call to `tools.subprocess()`. At this point. you're explicitly updating an empty environment, so an environment value of `None` is ignored.

So - this block can be simplified to:
```suggestion
        if overrides:
            env = overrides.copy()
        else:
            env = {}
```